clang/docs/HLSL/ExpectedDifferences.rst

```diff
@@ -0,0 +1,107 @@
+compilers are focused on increased consistency and correctness. Both reference
+compilers do not always apply language rules the same in all contexts.
+
+Clang also deviates from the reference compilers by providing different
+diagnostics, both in terms of the textual messages and the contexts in which
+diagnostics are produced. While striving for a high level of source
+compatibility with conforming HLSL code, Clang may produce earlier and more
+robust diagnostics for incorrect code or reject code that a reference compiler
+incorrectly accepted.
+
+Language Version
+================
+
+Clang targets language compatibility for HLSL 2021 as implemented by DXC.
+Language features that were removed in earlier versions of HLSL may be added on
+a case-by-case basis, but are not planned for the initial implementation.
+
+Overload Resolution
+===================
+
+Clang's HLSL implementation adopts C++ overload resolution rules as proposed for
+HLSL 202x based on proposal
+`0007 <https://github.com/microsoft/hlsl-specs/blob/main/proposals/0007-const-instance-methods.md>`_
+and
+`0008 <https://github.com/microsoft/hlsl-specs/blob/main/proposals/0008-non-member-operator-overloading.md>`_.
+
+Clang's implementation extends standard overload resolution rules to HLSL
+library functionality. This causes subtle changes in overload resolution
+behavior between Clang and DXC. Some examples include:
+
+.. code-block:: c++
+
+  void halfOrInt16(half H);
+  void halfOrInt16(uint16_t I);
+
+  void takesDoubles(double, double, double);
+
+  cbuffer CB {
+    uint U;
+    int I;
+    float X, Y, Z;
+    double3 A, B;
+  }
+
+  export void call() {
+    halfOrInt16(U); // All: Resolves to halfOrInt16(uint16_t).
+    halfOrInt16(I); // All: Resolves to halfOrInt16(uint16_t).
+    half H;
+  #ifndef IGNORE_ERRORS
+    H = asfloat16(I); // DXC: Fails to resolve overload for int.
+                      // Clang: Resolves to asfloat16(uint16_t).
+    H = asfloat16(U); // DXC: Fails to resolve overload for int.
+                      // Clang: Resolves to asfloat16(uint16_t).
+  #endif
+    H = asfloat16(0x01); // DXC: Resolves to asfloat16(half).
+                         // Clang: Resolves to asfloat16(uint16_t).
+
+    takesDoubles(X, Y, Z); // Works on all compilers
+  #ifndef IGNORE_ERRORS
+    fma(X, Y, Z); // DXC: Fails to resolve no known conversion from float to double.
+                  // Clang: Resolces to fma(double,double,double).
+  #endif
+    
+    double D = dot(A, B); // DXC: Resolves to dot(double3, double3), fails DXIL Validation.
+                          // FXC: Expands to compute double dot product with fmul/fadd
+                          // Clang: Resolves to dot(float3, float3), emits conversion warnings.
+
+  }
+
+.. note::
+
+  In Clang a conscious decision was made to exclude the ``dot(vector<double,N>,
+  vector<double,N>)`` overload and allow overload resolution to resolve the
```

Comment:
```suggestion
  In Clang a conscious decision was made to exclude the ``dot(vector<double,N>, vector<double,N>)`` 
  overload and allow overload resolution to resolve the
```
 
The new-line splitting the verbatim text is rendered as such in the rendered version. Suggest a fix as above.